services/controllerhost/extentmon.go

```diff
@@ -524,6 +526,41 @@ func (monitor *extentStateMonitor) handleDestinationExtent(dstDesc *shared.Desti
 			monitor.fixOutOfSyncStoreExtents(dstDesc.GetDestinationUUID(), extent)
 		}
 	}
+	
+	if common.IsRemoteZoneExtent(extent.GetOriginZone(), context.localZone) {
+		monitor.handleRemoteZoneDestinationExtent(dstDesc, isDLQ, extent)
+	}
+}
+
+// handleRemoteZoneDestinationExtent is the local handler for remote destination extent
+// this handler kicks off an event if the primary replicator for the extent is down
+func (monitor *extentStateMonitor) handleRemoteZoneDestinationExtent(dstDesc *shared.DestinationDescription, isDLQ bool, extent *metadata.DestinationExtent) {
+	context := monitor.context
+	// handle remote zone extent replication job failures
+	var failedStores []string
+	for _, storeID := range extent.GetStoreUUIDs() {
+		state, duration := context.failureDetector.GetHostState(common.StoreServiceName, storeID)
+		switch state {
+		case dfddHostStateUnknown:
+			failedStores = append(failedStores, storeID)
```

Comment:
let's say a store went down and immediately after the controller restarted. And let's say, the store that went down earlier never comes back online. In that case, controller will never *discover* that host, so the host state will always be unknown.